packages/jellyfish-api-core/__tests__/category/oracle/listPrices.test.ts

```diff
@@ -0,0 +1,97 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import BigNumber from 'bignumber.js'
+
+describe('Oracle', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should listPrices', async () => {
+    const priceFeeds1 = { token: 'APPLE', currency: 'EUR' }
+    const priceFeeds2 = { token: 'TESLA', currency: 'USD' }
+
+    const oracleid1 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeeds1], 1])
+    const oracleid2 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeeds1], 2])
+    const oracleid3 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeeds2], 3])
+    const oracleid4 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeeds2], 4])
+
+    await container.generate(1)
+
+    const timestamp = Math.floor(new Date().getTime() / 1000)
+
+    const prices1 = [{ tokenAmount: '0.5@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid1, timestamp, prices1])
+
+    const prices2 = [{ tokenAmount: '1.0@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid2, timestamp, prices2])
+
+    const prices3 = [{ tokenAmount: '1.5@TESLA', currency: 'USD' }]
+    await container.call('setoracledata', [oracleid3, timestamp, prices3])
+
+    const prices4 = [{ tokenAmount: '2.0@TESLA', currency: 'USD' }]
+    await container.call('setoracledata', [oracleid4, timestamp, prices4])
+
+    await container.generate(1)
+
+    const data = await client.oracle.listPrices()
+
+    // NOTE(jingyi2811): 0.83333333000000 = (0.5 * 1 + 1 * 2) / 3
+    // NOTE(jingyi2811): 1.78571428000000 = (1.5 * 3 + 2 * 4) / 7
+    expect(data).toStrictEqual([
+      { token: 'APPLE', currency: 'EUR', price: new BigNumber(0.83333333000000), ok: true },
+      { token: 'TESLA', currency: 'USD', price: new BigNumber(1.78571428000000), ok: true }
+    ])
+
+    await container.call('removeoracle', [oracleid1])
+    await container.call('removeoracle', [oracleid2])
+    await container.call('removeoracle', [oracleid3])
+    await container.call('removeoracle', [oracleid4])
```

Comment:
move removeoracle logic into `afterEach()`, you it needs to cleanup after itself. Apply this test pattern to all your oracles tests if you haven't already. Since your test needs a clean state.

```ts
afterEach(() => {
 // get list or oracles
 // remove all of them
})
```

---

This is instead of doing this: https://github.com/DeFiCh/jellyfish/commit/5f4537cc1912093b8bdb649ec4f129529fb6c00f

Where you need a clean container for all your tests, you can remove the oracle at afterEach.

Reply:
I have applied this to all the rpcs of both getPrice and listPrices branches